src/app/components/shell/router.js

```diff
@@ -126,6 +126,16 @@ function RedirectToCanonicalDetailsPage() {
     );
 }
 
+function RedirectToPageWithoutJunk() {
+    const {name, '*': junk} = useParams();
```

Comment:
this ditches everything other than the top url section? so `/foo/bar` will be redirected to `/foo` ? should we not just 404?

Reply:
We can, but I don't know why or how often we're getting URLs like `/impact/openstax-has-been-a-great-teacher`. I would think it's a better user experience to go to the obvious page.

Reply:
I have added a commit that 404s instead.